pallets/liquidity-rewards/src/lib.rs

```diff
@@ -196,6 +196,10 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	// Although this value could be stored inside `EpochData`,
+	// we maintain it separately to avoid deserializing the whole EpochData struct each `on_initialize()` call.
+	// EpochData could be relatively big if there many groups.
+	// We dont have to deserialize the whole struct 99% of the time assuming a duration of 100 blocks.
```

Comment:
Very informative comment! I just don't quite get the last sentence. Can you explain?

Reply:
Michael's Suggestion 😄 

Yes, for each `on_initialize()` call, you need to check the final timestamp of an epoch to check if the epoch has finalized, but only when the epoch finalizes do you need to read its content. If it happens every 100 blocks (for example), we only deserialize it in that case, in the last block of those 100 (1% of the cases).

Reply:
I would add this explanation to this comment block tbh :)